### PR TITLE
[JUJU-1150]Remove feature flag added for syslog log-sink

### PR DESCRIPTION
**Pull request by basebandit**
This PR reverts this [one](https://github.com/juju/juju/pull/13971) and removes the logoutput feature flag. The feature flag was added to the `2.9` branch in order to require a specific opt-in action for using Syslog as a log-sink instead of MongoDB and brought forward to `develop`. We want to remove the flag from the develop branch, as it will become a standard feature. When we move to a SQL-based back-end, it will be the only log-sink option.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- ~[ ] Comments saying why design decisions were made~
- ~[ ] Go unit tests, with comments saying what you're testing~
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```sh
$ juju bootstrap aws/eu-west-1 test --config "logging-output=syslog"
  Creating Juju controller "aws-eu-west-1" on aws/eu-west-1
  ...
$ juju bootstrap aws test --config "logging-output=syslog,database"
$ juju add-model foo --config logging-output=syslog
$ juju model-config logging-output=syslog
```

### CODE REVIEW

File: apiserver/facades/client/modelconfig/modelconfig_test.go [hunk 1]
```diff
@@ -5,22 +5,19 @@ package modelconfig_test
...
-	"github.com/juju/juju/feature"
 	"github.com/juju/juju/provider/dummy"
 	_ "github.com/juju/juju/provider/dummy"
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
 	"github.com/juju/juju/testing"
+	"github.com/juju/names/v4"
+	gitjujutesting "github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
```
Comment (anvial) [inline apiserver/facades/client/modelconfig/modelconfig_test.go:20]: This import order is not correct, please read the Import section in https://github.com/juju/juju/blob/develop/CONTRIBUTING.md

Comment (basebandit) [reply apiserver/facades/client/modelconfig/modelconfig_test.go:20]: The lint did not complain. Might have to do it manually. Thank you.
```
./main.sh static_analysis test_static_analysis_go

==> Checking for dependencies
==> Running subtest test_static_analysis_go for static_analysis suite
==> TEST BEGIN: static_analysis (tmp.9JE)
==> TEST SKIPPED: static copyright analysis
==> TEST SKIPPED: test doc go
===> [   ] Running: go
===> [ ✔ ] Success: go (19s)
===> [   ] Running: go tidy
===> [ ✔ ] Success: go tidy (0s)
==> TEST SKIPPED: static shell analysis
==> TEST SKIPPED: static python analysis
==> TEST SKIPPED: static schema analysis
==> TEST DONE: static_analysis (19s)
==> Cleaning up
====> Completed cleaning up jujus

==> Test result: success
==> Tests Removed: /home/capricorn/code/go/src/juju/tests/tmp.9JE
==> TEST COMPLETE
```

File: apiserver/facades/client/modelconfig/modelconfig_test.go [hunk 2]
```diff
@@ -4,23 +4,21 @@
...
-	"github.com/juju/juju/controller"
 	"github.com/juju/juju/core/constraints"
 	"github.com/juju/juju/environs/config"
-	"github.com/juju/juju/feature"
 	"github.com/juju/juju/provider/dummy"
 	_ "github.com/juju/juju/provider/dummy"
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
 	"github.com/juju/juju/testing"
+	"github.com/juju/names/v4"
```
Comment (manadart) [inline apiserver/facades/client/modelconfig/modelconfig_test.go:19]: Import ordering again.

**Review by manadart — APPROVED** [review]
Still some import groupings to be fixed, but otherwise looks OK.

File: apiserver/facades/client/modelconfig/modelconfig_test.go (hunk 1 above)
Comment (basebandit) [reply apiserver/facades/client/modelconfig/modelconfig_test.go]: Fixed manually

Comment (anvial) [reply apiserver/facades/client/modelconfig/modelconfig_test.go]: Thx, now looks good

**Review by anvial — APPROVED** [review]

**Pull request merged**